<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Data Summary</title>

<script src="site_libs/header-attrs-2.28/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html"></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="proposal.html">Proposal</a>
</li>
<li>
  <a href="report.html">Report</a>
</li>
<li>
  <a href="introduction.html">Introduction</a>
</li>
<li>
  <a href="data.html">Data</a>
</li>
<li>
  <a href="eda.html">EDA</a>
</li>
<li>
  <a href="analysis.html">Analysis</a>
</li>
<li>
  <a href="discussion.html">Discussion</a>
</li>
<li>
  <a href="appendix.html">Appendix</a>
</li>
<li>
  <a href="mailto:&lt;dl3757@cumc.columbia.edu&gt;">
    <span class="fa fa-play-circle fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="mailto:&lt;dl3757@cumc.columbia.edu&gt;">
    <span class="fa fa-envelope fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://github.com/DangLin1214/test4.github.io.git">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Data Summary</h1>

</div>


<div id="ukb-dataset" class="section level3">
<h3>UKB Dataset</h3>
<p>The dataset used in this project is the UKB dataset. The UK Biobank
provides a comprehensive dataset from 500,000 participants aged 40–69,
collected between 2006 and 2010, to support health research. It includes
baseline data from physical assessments, questionnaires, and health
records, alongside genetic information from genotyping and sequencing.
The resource also features imaging data (e.g., MRI and DEXA scans),
biological samples (blood, urine, and saliva), linked health records
(hospital admissions, primary care, and disease registries), activity
data from wearable devices, and environmental information on air quality
and socioeconomic factors. This extensive dataset is designed to advance
understanding, prevention, diagnosis, and treatment of a wide range of
diseases.</p>
<p>After the data cleaning steps, the cleaned dataset includes 413,146
observations and 113 variables. Although we will not use all the
variables in this dataset, we will select multiple key variables to
investigate our research questions:</p>
<p><code>expo_npar_cat</code>: Neutrophil-to-albumin ratio categorized
into quartiles (Q1 = lowest, Q4 = highest). <br></p>
<p><code>age</code>: Participants’ age, categorized into 10-year age
groups (e.g., 30–39, 40–49, etc.). <br></p>
<p><code>sex</code>: Male or female. <br></p>
<p><code>income</code>: Seven income brackets, ranging from 1 (lowest)
to 7 (highest). <br></p>
<p><code>townsend</code>: Townsend deprivation index, a measure of
socioeconomic status, where higher values indicate greater deprivation.
<br></p>
<p><code>total_met</code>: Total metabolic equivalent of task score,
derived from physical activity data and grouped into categories.
<br></p>
<p><code>diet_quality</code>: Diet quality score, ranging from 0
(poorest) to 7 (highest), based on adherence to dietary guidelines.
<br></p>
<p><code>sleep_hour</code>: Nightly sleep duration, measured in hours.
<br></p>
<p><code>smoke_cat</code>: Smoking status, categorized as current
smoker, past smoker, or non-smoker. <br></p>
<p><code>total_alcohol</code>: Total alcohol consumption, grouped into
categories (non-drinker, low, moderate, high, very high). <br></p>
<p><code>non_hdl</code>: Non-HDL cholesterol level, a continuous
variable indicating cardiovascular risk. <br></p>
<p><code>tg</code>: Triglyceride levels, a continuous measure of blood
lipid levels. <br></p>
<p><code>bp_cat</code>: Blood pressure, categorized into intervals
(e.g., BP=20, 40, 60, etc.). <br></p>
<p><br></p>
<div id="exposure-variables" class="section level4">
<h4>Exposure Variables</h4>
<p><strong>Lymphocyte-to-Monocyte Ratio (LMR):</strong> <span
class="math display">\[\text{LMR} = \frac{\text{Absolute Lymphocyte
Count}}{\text{Absolute Monocyte Count}}\]</span></p>
<p><br></p>
<p><strong>Systemic Immune-Inflammation Index (SII):</strong> <span
class="math display">\[\text{SII} = \frac{\text{Neutrophil Count} \times
\text{Platelet Count}}{\text{Lymphocyte Count}}\]</span></p>
<p><br></p>
<p><strong>Naples Prognostic Score (NPS):</strong> This score is based
on a combination of factors, typically including:</p>
<ul>
<li>Albumin level (&lt;4 g/dL = 1 point).</li>
<li>Neutrophil-to-Lymphocyte Ratio (NLR) (&gt;3 = 1 point).</li>
<li>Lymphocyte-to-Monocyte Ratio (LMR) (&lt;4.44 = 1 point).</li>
<li>Platelet-to-Lymphocyte Ratio (PLR) (&gt;150 = 1 point).</li>
</ul>
<p>The final NPS is the sum of these points (range: 0–4).</p>
<p><br></p>
<p><strong>Neutrophil-Percentage-to-Albumin Ratio (NPAR):</strong> <span
class="math display">\[\text{NPAR} = \frac{\text{Neutrophil
Percentage}}{\text{Albumin Level (g/dL)}}\]</span></p>
<p><br></p>
</div>
</div>
<div id="data-cleaning" class="section level3">
<h3>Data Cleaning</h3>
<div id="data-cleaning-part-one" class="section level4">
<h4>Data Cleaning Part One</h4>
<p><a
href="https://github.com/DangLin1214/test4.github.io/blob/main/pre_cleaning.md">View
Full Part One Data Cleaning Steps on GitHub</a></p>
<p>The pre-cleaning process involved several key steps to transform the
raw UKB dataset into a clean, analyzable format. Initially, variables
were renamed to improve clarity and usability. For instance,
<code>n_21022_0_0</code> was renamed to <code>age</code>,
<code>n_31_0_0</code> to <code>sex</code>, and <code>n_21000_0_0</code>
to <code>ethnicity</code>. New variables were derived from existing data
to enhance analysis potential. For example, physical activity metrics
such as <code>MET_walk</code> (<code>n_22037_0_0</code>),
<code>MET_moderate</code> (<code>n_22038_0_0</code>), and
<code>MET_vigorous</code> (<code>n_22039_0_0</code>) were combined to
create <code>total_MET</code>, which was further categorized into
activity levels using <code>pa_cat</code>. Invalid or missing data
values, such as <code>-3</code> and <code>-1</code>, were replaced with
<code>.</code> (missing), while specific imputations were applied where
necessary, such as assigning <code>0.5</code> to dietary variables like
<code>cooked_vegetable_0</code> when valid data was unavailable.</p>
<p>Variables were categorized for analytical ease, such as grouping
physical activity, sleep, and dietary patterns. Dietary components like
vegetables, fruits, fish, red meat, and grains were processed to create
binary indicators of “healthy” consumption (e.g.,
<code>healthy_vegetable</code> for ≥3 servings/day of vegetables and
<code>healthy_red_meat</code> for ≤1.5 servings/week of red meat). An
aggregate dietary quality score (<code>diet_quality</code>) was computed
by summing these indicators and further categorized into quality
levels.</p>
<p>For sleep patterns, multiple variables such as chronotype
(<code>Chronotype_preference_0</code>), insomnia
(<code>Insomnia_0</code>), snoring (<code>snoring_0</code>), and daytime
sleepiness (<code>daytime_sleepiness_0</code>) were processed to
calculate a composite sleep quality score (<code>sleep_quality</code>).
This score was then combined with categorized sleep duration
(<code>sleep_cat</code>) to refine the assessment of overall sleep
health.</p>
<p>Lifestyle factors like smoking and alcohol consumption were also
carefully processed. Smoking history variables
(<code>smoke_current</code> and <code>smoke_past</code>) were combined
to categorize individuals’ smoking behavior into <code>smoke_cat</code>.
Alcohol consumption was calculated using variables like
<code>redwine_week</code>, <code>beer_week</code>, and
<code>spirits_week</code>, and total intake was categorized into risk
levels.</p>
<p>Biomarkers such as lipid and glucose levels were processed to
generate new variables like <code>non_hdl</code> (non-HDL cholesterol
derived from <code>tc</code> and <code>hdl</code>) and
<code>glu_cat</code> (glucose levels categorized based on clinical
thresholds). Blood pressure variables (<code>sbp</code> and
<code>dbp</code>) were similarly used to classify individuals into
health risk categories.</p>
<p>The cleaned and processed dataset, containing all these refined and
derived variables, was saved as a new csv to be used for further
analysis. This comprehensive pre-cleaning ensured that the dataset was
both reliable and tailored to support the research objectives.</p>
<p><br></p>
</div>
<div id="data-cleaning-part-two" class="section level4">
<h4>Data Cleaning Part Two</h4>
<p><a
href="https://github.com/DangLin1214/test4.github.io/blob/main/data_preparation.md">View
Full Part Two Data Cleaning Steps on GitHub</a></p>
<p>The subsequent data cleaning process was designed to ensure the
dataset’s quality and readiness for complex analyses, including
multi-state modeling. The process began with importing the csv file
exported in the previous steps and standardizing variable names using
<code>janitor::clean_names()</code> to improve readability and
usability. Key variables were then transformed and categorized. For
instance, <code>sex</code> was converted into a factor variable with 0
representing female and 1 representing male, while
<code>ethnicity_cat</code> was categorized as 0 for white and 1 for
others. Age was grouped into meaningful categories
(<code>age_cat</code>): <code>&lt;50</code>, <code>50–59</code>,
<code>60–69</code>, and <code>≥70</code>. Socioeconomic variables, like
<code>income</code>, and lifestyle variables, such as smoking status
(<code>smoke_cat</code>), were also transformed into categorical
factors.</p>
<p>Participants with pre-existing conditions at baseline were excluded
to ensure accurate survival and progression analyses. This included
those with non-alcoholic fatty liver disease (NAFLD) or cirrhosis,
identified through the variables <code>nafld_date</code> and
<code>cirrho_date</code>. Outcome variables were created to classify
participants’ health status. For example, <code>nafld_outcome</code> and
<code>cirrho_outcome</code> assigned 0 for healthy participants and 1
for those who developed the disease post-baseline. Missing disease dates
were carefully imputed using other known dates, such as the death date
(<code>death_date</code>) or lost-to-follow-up date
(<code>lost_date</code>). For participants with no recorded outcomes, a
default date of <code>2022-10-31</code> was used to mark the end of
follow-up.</p>
<p>To calculate disease progression metrics, survival durations were
computed from the baseline to either disease diagnosis or the end of
follow-up. These metrics were stored as <code>nafld_surv_duration</code>
and <code>cirrho_surv_duration</code>. Blood test variables were also
renamed and cleaned, including lymphocytes (<code>b_lympho</code>),
monocytes (<code>b_mono</code>), platelets (<code>b_plate</code>), and
neutrophils (<code>b_neutro</code>). Observations with missing values in
these variables were dropped to ensure reliable downstream analyses.</p>
<p>New biomarkers were derived to assess participants’ health status and
inflammation levels. For example, the lymphocyte-to-monocyte ratio
(<code>expo_lmr</code>), systemic immune-inflammation index
(<code>expo_sii</code>), and neutrophil-to-albumin ratio
(<code>expo_npar</code>) were calculated. Critical thresholds were
applied to categorize these biomarkers into clinical risk groups, such
as low albumin levels (<code>cri_album</code> &lt;4 g/dL) or high
neutrophil-to-lymphocyte ratios (<code>cri_nlr</code> &gt;3). Biomarkers
were further categorized into quartiles using a custom function
(<code>cut_1_23_4</code>) to allow for stratified analyses.</p>
<p>The final dataset included essential variables, such as demographics
(<code>sex</code>, <code>age</code>), lifestyle factors
(<code>smoke_cat</code>, <code>total_alcohol</code>), biomarkers (e.g.,
<code>tc</code> for total cholesterol, <code>hdl</code> for high-density
lipoprotein, <code>hba1c</code> for blood glucose), and disease
outcomes. These variables were selected and exported as
<code>data_prepared.dta</code>. For advanced analyses, such as
multi-state modeling, the dataset was structured to capture transitions
between disease states (Healthy → NAFLD → Cirrhosis → Death), including
time intervals and durations. This ensured the dataset was robust,
consistent, and ready for both descriptive and inferential analyses,
particularly for modeling disease progression pathways.</p>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
